pkg/router/routeapihelpers/validation.go

```diff
@@ -367,3 +367,32 @@ func validateCertificatePEM(certPEM string, options *x509.VerifyOptions) ([]*x50
 
 	return certs, nil
 }
+
+// DeprecatedRouteValidation performs an upgrade validation for
+// a route. This checks for issues that will cause failures in the next
+// OpenShift version.
+func DeprecatedRouteValidation(route *routev1.Route) field.ErrorList {
+	tlsConfig := route.Spec.TLS
+	result := field.ErrorList{}
+
+	if tlsConfig == nil {
+		return result
+	}
+
+	tlsCertFieldPath := field.NewPath("spec").Child("tls").Child("certificate")
+	if len(tlsConfig.Certificate) > 0 {
+		certs, err := cert.ParseCertsPEM([]byte(tlsConfig.Certificate))
+		if err != nil {
+			return result
+		}
```

Comment:
I think you can do this `return result` (where result is an empty list of errors) because other validation will report errors from `ParseCertsPEM`, right?  Please add a comment explaining this reasoning.

Reply:
Done.